text/0314-cdk-construct-hub.md

```diff
@@ -0,0 +1,278 @@
+   artifacts consumed by the front-end application, stored in a dedicated S3
+   bucket using the following key format:
+   `assemblies/${assembly.name}/v${assembly.version}/assembly.json`
+
+   This function validates the contents of the `assembly` to ensure the message
+   was well-formed. In case the message is found to be incoherent, it is sent to
+   a dead-letter queue and an alarm is triggered.
+
+   It also creates or updates the `assemblies/${assembly.name}/versions.json`
+   object, which contains a list of all versions ever indexed for the package,
+   matched to the versions' current status (`deprecated`, `latest`, ...):
+
+   Name        | Description
+   ------------|----------------------------------------------------------------
+   `v${major}` | All versions part of this major version stream, and their status
+
+   [SemVer]: https://semver.org/spec/v2.0.0.html
+
+1. A series of Lambda functions prepare language-specific assembly files for
+   each configured language, with adjusted naming conventions, and updated
+   sample code fragments, and proceeds to store those at:
+   `assemblies/${assembly.name}/v${assembly.version}/assembly-${lang}.json`.
+   This transformation is backed by the `jsii-rosetta` tool, which is part of
+   the [jsii project][jsii].
+
+   [jsii]: https://aws.github.io/jsii
+
+1. A Lambda function keeps the `latest.json` object updated with the last 20
+   package versions indexed (according to the `time` field reported by the
+   ingestion component).
+
+1. A Lambda function keeps the `catalog.json` object updated with the latest
+   versions of each package's major version lines, which backs the websites'
+   search page. The object may be sharded if it becomes too large.
+
+#### Front-End
+
+The front-end consists in a simple [React] application hosted in an S3 bucket.
+A CloudFront distribution fronts this application, ensuring it is served from
+edge locations close to any customer, and binding together the static assets
+that compose the application and the data objects maintained by the back-end
+pipeline.
+
+1. The landing page asynchronously loads the `latest.json` object to render the
+   "latest" tiles, containing the short-list of packages that changed most
+   recently.
+
+1. The search feature asynchronously loads the `catalog.json` object, and
+   performs client-side filtering to prepare search results.
+
+1. Package detail pages fetch the relevant `assembly.json` object from the
+   assembly store, and renders the detail page using information contained
+   therein.
+
+   - If a different language (than JavaScript) is selected, the
+     `assembly-${lang}.json` object is fetched instead, providing the relevant
+     documentation elements.
+
+[react]: https://reactjs.org
+
+### Are there any open issues that need to be addressed later?
+
+- When the catalog becomes too large to be convenient to process in-memory on
```

Comment:
Should we show some data based on the current catalog size? See [packages.json](https://awscdk.io/index/packages.json)